pages/_app.js

```diff
@@ -54,7 +54,7 @@ function MyApp ({ Component, pageProps: { ...props } }) {
     router.replace({
       pathname: router.pathname,
       query: { ...router.query, nodata: true }
-    }, router.asPath, { ...router.options, shallow: true }).catch((e) => {
+    }, props.me ? `?r=${props.me.name}` : router.asPath, { ...router.options, shallow: true }).catch((e) => {
```

Comment:
Hmmm I'm not sure this is right. Doesn't this make the url bar always `stacker.news?r=<nym>` regardless of what page we're on? 

I think we want to "add" the query parameter to `router.asPath`

Reply:
It adds the param to the url like this as examples:
- root: `stacker.news/~?r=username`
- top: `stacker.news/~/top/posts/day?r=username`
- recent: `stacker.news/~/recent?r=username`
- item: `stacker.news/items/4?r=username`

Reply:
I see. I would like to keep our url pattern otherwise the same while introducing this change.

If you'd like I can try to take this the last yard on my own. You've done the lionshare of the effort already.

> It adds the param to the url like this as examples:
> - root: stacker.news/~?r=username
> - top: stacker.news/~/top/posts/day?r=username
> - recent: stacker.news/~/recent?r=username
> - item: stacker.news/items/4?r=username

This is what ideally we'd get instead:
- root: stacker.news?r=username
- top: stacker.news/top/posts/day?r=username
- recent: stacker.news/recent?r=username
- item: stacker.news/items/4?r=username

Reply:
Ah, I see. Without the tilde added. I don't actually know where that's coming from. I see it's apart of the pages:

```sh
└── ~
    ├── index.js
    ├── post.js
    ├── recent
    │   ├── [type].js
    │   └── index.js
    ├── rss.js
    ├── search.js
    └── top
        ├── [type]
        │   └── [when].js
        ├── cowboys.js
        └── stackers
            └── [when].js
```

I don't know why my changes are forcing it though.

Either way, go ahead and take it to the finish line. I don't think I'll be able to find time to work on it this coming week. But, now I'm that much more familiar with the codebase for next time.

Reply:
Cool, will do!